packages/server/src/automations/actions.ts

```diff
@@ -20,37 +20,41 @@ import * as triggerAutomationRun from "./steps/triggerAutomationRun"
 import env from "../environment"
 import {
   AutomationStepSchema,
-  AutomationStepInput,
   PluginType,
   AutomationStep,
+  AutomationActionStepId,
+  AutomationStepInputMap,
+  AutomationStepInput,
 } from "@budibase/types"
 import sdk from "../sdk"
 import { getAutomationPlugin } from "../utilities/fileSystem"
 
-const ACTION_IMPLS: Record<
-  string,
-  (opts: AutomationStepInput) => Promise<any>
-> = {
-  SEND_EMAIL_SMTP: sendSmtpEmail.run,
-  CREATE_ROW: createRow.run,
-  UPDATE_ROW: updateRow.run,
-  DELETE_ROW: deleteRow.run,
-  OUTGOING_WEBHOOK: outgoingWebhook.run,
-  EXECUTE_SCRIPT: executeScript.run,
-  EXECUTE_QUERY: executeQuery.run,
-  SERVER_LOG: serverLog.run,
-  DELAY: delay.run,
-  FILTER: filter.run,
-  QUERY_ROWS: queryRow.run,
-  COLLECT: collect.run,
-  TRIGGER_AUTOMATION_RUN: triggerAutomationRun.run,
+const ACTION_IMPLS: Partial<{
+  [K in keyof AutomationStepInputMap]: (
+    opts: AutomationStepInput<K>
```

Comment:
I was finding it really hard to get this actually working properly and I'm not sure if this makes any sense / is actually useful?

Reply:
It is, but it might change the keys. It will not be `ACTION_IMPLS.CREATE_ROW`, instead it will change to the string value of the `AutomationActionStepId`. In this case it works fine because they have the same string value as the key, but it might break.
You might want to use: `{[key in keyof typeof AutomationActionStepId]:  (opts: AutomationStepInput<key>) => Promise<any> }`